src/Microsoft.ML.TimeSeries/SsaSpikeDetector.cs

```diff
@@ -122,5 +167,125 @@ public override void Save(ModelSaveContext ctx)
 
             base.Save(ctx);
         }
+
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+    }
+
+    public sealed class SsaSpikeEstimator : IEstimator<ITransformer>
+    {
+        private readonly IHost _host;
+        private readonly string _inputColumnName;
+        private readonly string _outputColumnName;
+        private readonly int _confidence;
+        private readonly int _pvalueHistoryLength;
+        private readonly int _trainingWindowSize;
+        private readonly int _seasonalityWindowSize;
+
+        public SsaSpikeEstimator(
+            IHostEnvironment env,
+            int confidence,
+            int pvalueHistoryLength,
+            int trainingWindowSize,
+            int seasonalityWindowSize,
+            string input,
+            string output)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register("SsaSpikeEstimator");
+
+            _host.CheckNonEmpty(input, nameof(input));
+            _host.CheckNonEmpty(output, nameof(output));
+
+            _confidence = confidence;
+            _pvalueHistoryLength = pvalueHistoryLength;
+            _trainingWindowSize = trainingWindowSize;
+            _seasonalityWindowSize = seasonalityWindowSize;
+            _inputColumnName = input;
+            _outputColumnName = output;
+        }
+
+        public ITransformer Fit(IDataView input)
+        {
+            _host.CheckValue(input, nameof(input));
+            var transformer = new SsaSpikeDetector(_host,
+                new SsaSpikeDetector.Arguments
+                {
+                    Confidence = _confidence,
+                    PvalueHistoryLength = _pvalueHistoryLength,
+                    TrainingWindowSize = _trainingWindowSize,
+                    SeasonalWindowSize = _seasonalityWindowSize,
+                    Source = _inputColumnName,
+                    Name = _outputColumnName
+                });
+            transformer.Fit(input);
+            return transformer;
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var resultDic = inputSchema.Columns.ToDictionary(x => x.Name);
+
+            resultDic[_outputColumnName] = new SchemaShape.Column(
+                _outputColumnName, SchemaShape.Column.VectorKind.Vector, NumberType.R8, false);
+
+            return new SchemaShape(resultDic.Values);
+        }
+    }
+
+    public static class SsaSpikeStaticExtensions
+    {
+        private sealed class OutColumn : Vector<float>
+        {
+            public PipelineColumn Input { get; }
+
+            public OutColumn(Vector<float> input,
```

Comment:
>OutColumn [](start = 19, length = 9)

Did you put static method which calls this OutColumn in other class, or it just missing?

Reply:
removed pigsty extensions, will be part of separate PR

---
In reply to: [225356515](https://github.com/dotnet/machinelearning/pull/1254#discussion_r225356515) [](ancestors = 225356515)